src/app/habits/createhabit.tsx

```diff
@@ -0,0 +1,238 @@
+            placeholderTextColor={DefaultColors[colorScheme].placeholder}
+            value={habitName}
+            onChangeText={(text) => setHabitName(text)}
+          ></TextInput>
+        </View>
+      </View>
+
+      {/* Description */}
+      <View className="flex flex-col gap-y-2">
+        <Text className="text-base font-semibold">Description (optional)</Text>
+        <TextInput
+          className="h-10 rounded-lg border px-3"
+          style={{
+            borderColor,
+            color: DefaultColors[colorScheme].text,
+          }}
+          maxLength={50}
+          value={description}
+          onChangeText={(text) => setDescription(text)}
+        ></TextInput>
+      </View>
+
+      {/* Colour */}
+      <View className="flex flex-col gap-y-1">
+        <Text className="text-base font-semibold">Color</Text>
+        <View className="flex flex-row flex-wrap justify-between gap-x-1 gap-y-1">
+          {colorNames.map((col, idx) => (
+            <TouchableOpacity
+              className="flex h-9 w-9 items-center justify-center rounded-full"
+              key={idx}
+              style={{
+                backgroundColor:
+                  col === "stone"
+                    ? colors.stone[400]
+                    : colors.habitColors[col]?.base,
+              }}
+              onPress={() => {
+                setColor(col);
+              }}
+            >
+              {col === color ? <Icon icon={IconCheck} /> : null}
+            </TouchableOpacity>
+          ))}
+        </View>
+      </View>
+
+      {/* Goal */}
+      <View className="flex flex-col gap-y-2">
+        <Text className="text-base font-semibold">Goal</Text>
+        <View className="rounded-lg border" style={{ borderColor }}>
+          <View
+            className="flex flex-row justify-between p-2"
+            style={{ backgroundColor: colors.transparent }}
+          >
+            <Text className="text-base font-semibold">Goal type</Text>
+            <Menu onSelect={(val) => setGoalType(val)}>
+              <MenuTrigger>
+                <View className="flex flex-row">
+                  <Text className="text-base text-stone-400">
+                    {goalType} goal
+                  </Text>
+                  <Icon
+                    icon={IconSelector}
+                    darkColor={colors.stone[400]}
+                    lightColor={colors.stone[400]}
+                  />
+                </View>
+              </MenuTrigger>
+              <MenuOptions>
+                <MenuOption value="Daily" text="Daily goal" />
+                <MenuOption value="Weekly" text="Weekly goal" />
+              </MenuOptions>
+            </Menu>
+          </View>
+          <Divider />
+          <View
+            className="flex flex-row justify-between p-2"
+            style={{ backgroundColor: colors.transparent }}
+          >
+            <Text className="text-base font-semibold">{goalType} goal</Text>
```

Comment:
I just changed the Figma; I think it would be better to say "Completions per day" (or per week) instead of the current way